contracts/masset/liquidator/Liquidator.sol

```diff
@@ -219,57 +304,207 @@ contract Liquidator is ILiquidator, Initializable, ImmutableModule {
+            IUniswapV3SwapRouter.ExactInputParams(
+                liquidation.uniswapPath,
+                address(this),
+                block.timestamp,
+                sellAmount,
+                minOut
+            );
+        uniswapRouter.exactInput(param);
+
+        // 4. Mint mAsset using purchased bAsset
+        address mAsset = liquidation.mAsset;
+        uint256 minted = _mint(bAsset, mAsset);
+
+        // 5. Send to SavingsManager
+        address savings = _savingsManager();
+        ISavingsManager(savings).depositLiquidation(mAsset, minted);
 
-        // 3.3. Trade on Curve
-        uint256 purchased = _sellOnCrv(bAsset, liquidation.curvePosition);
+        emit Liquidated(sellToken, mAsset, minted, bAsset);
+    }
 
-        // 4.0. Send to SavingsManager
-        address savings = _savingsManager();
-        IERC20(mUSD).safeApprove(savings, 0);
-        IERC20(mUSD).safeApprove(savings, purchased);
-        ISavingsManager(savings).depositLiquidation(mUSD, purchased);
+    /**
+     * @notice Claims stake Aave token rewards from each Aave integration contract
+     * and then transfers all reward tokens to the liquidator contract.
+     * Can only claim more stkAave if the last claim's unstake window has ended.
+     */
+    function claimStakedAave() external {
+        // solium-disable-next-line security/no-tx-origin
+        require(tx.origin == msg.sender, "Must be EOA");
+
+        // If the last claim has not yet been liquidated
+        uint256 totalAaveBalanceMemory = totalAaveBalance;
+        if (totalAaveBalanceMemory > 0) {
+            // Check unstake period has expired for this liquidator contract
+            IStakedAave stkAaveContract = IStakedAave(stkAave);
+            uint256 cooldownStartTime = stkAaveContract.stakersCooldowns(address(this));
+            uint256 cooldownPeriod = stkAaveContract.COOLDOWN_SECONDS();
+            uint256 unstakeWindow = stkAaveContract.UNSTAKE_WINDOW();
+
+            // Can not claim more stkAave rewards if the last unstake window has not ended
+            // Wait until the cooldown ends and liquidate
+            require(
+                block.timestamp > cooldownStartTime + cooldownPeriod,
+                "Last claim cooldown not ended"
+            );
+            // or liquidate now as currently in the
+            require(
+                block.timestamp > cooldownStartTime + cooldownPeriod + unstakeWindow,
+                "Must liquidate last claim"
+            );
+            // else the current time is past the unstake window so claim more stkAave and reactivate the cool down
+        }
+
+        // 1. For each Aave integration contract
+        uint256 len = aaveIntegrations.length;
+        for (uint256 i = 0; i < len; i++) {
+            address integrationAdddress = aaveIntegrations[i];
+
+            // 2. Claim the platform rewards on the integration contract. eg stkAave
+            PAaveIntegration(integrationAdddress).claimRewards();
```

Comment:
Should probably refactor this to be an interface as discussed

Reply:
I started adding `claimRewards` to `IPlatformIntegration` but that caused too many problems with the older Compound and Aave V2 integrations. I ended up adding a new `IClaimRewards` interface to remove the dependency to `PAaveIntegration` which causes issues with different `Initializable` contracts in the flattener.